packages/component-base/src/define.js

```diff
@@ -19,21 +21,34 @@ export function defineCustomElement(CustomElement, version = '24.6.0-alpha8') {
   });
 
   if (CustomElement.experimental) {
-    const name = CustomElement.is.split('-').slice(1).join('-');
-    const featureFlagKey = `${dashToCamelCase(name)}Component`;
-    if (!window.Vaadin.featureFlags[featureFlagKey]) {
+    const featureFlagKey =
+      typeof CustomElement.experimental === 'string'
+        ? CustomElement.experimental
+        : `${dashToCamelCase(CustomElement.is.split('-').slice(1).join('-'))}Component`;
+
+    if (!window.Vaadin.featureFlags[featureFlagKey] && !experimentalMap[featureFlagKey]) {
       // Add setter to define experimental component when it's set to true
+      experimentalMap[featureFlagKey] = new Set();
+      experimentalMap[featureFlagKey].add(CustomElement);
+
       Object.defineProperty(window.Vaadin.featureFlags, featureFlagKey, {
```

Comment:
The idea is to define the property on the object, not re-define the object itself though.

Reply:
Good point. I wonder if using `Object.defineProperty` is still safe here since it behaves differently than flags defined with `window.Vaadin.featureFlags.myCustomFlag = true;`. Was mainly thinking if products like Copilot can read it properly:

![Screenshot 2024-11-18 at 14 12 20](https://github.com/user-attachments/assets/dc193603-b2fa-43ad-bbb5-f8c631f842c5)

Reply:
Right, I didn't consider this case. Do you have any ideas how to handle this? Maybe we could use `Proxy` which also allows to override `getOwnPropertyDescriptor` although it would make the logic more complicated.

Reply:
I looked into the Copilot codebase and didn't find cases where the featureFlagsobject is iterated. I'm ok with the current approach.